tests/integration/components/marker-layer-test.js

```diff
@@ -110,6 +110,32 @@ test('marker updates dragging', function(assert) {
    assert.equal(marker._layer.dragging.enabled(), false, 'marker dragging disabled');
 });
 
+test('marker dragging bound when icon changes', function(assert) {
+
+  this.set('markerCenter', locations.nyc);
+  this.set('draggable', true);
+  this.set('currentIconUrl', 'custom-url.png');
+  this.set('currentSize', 12);
+
+  this.render(hbs`
+    {{#leaflet-map zoom=zoom center=center}}
+      {{marker-layer location=markerCenter draggable=draggable
+        icon=(icon
+          iconUrl=currentIconUrl
+          iconSize=(point currentSize currentSize))}}
+    {{/leaflet-map}}
+  `);
+
+    //pre-conditions
+   assert.ok(marker._layer.dragging.enabled(), 'marker dragging enabled');
+
+   this.set('draggable', false);
+   this.set('currentIconUrl', 'another-custom-url.png');
+
+   assert.equal(marker._layer.options.icon.options.iconUrl, 'another-custom-url.png');
+   assert.equal(marker._layer.dragging.enabled(), false, 'marker dragging disabled');
+});
```

Comment:
Just to make sure your using `enabled()` and not `enable()`: http://leafletjs.com/reference.html#ihandler-enabled